conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/bitcoin/BitcoinIndexer.scala

```diff
@@ -0,0 +1,92 @@
+package tech.cryptonomic.conseil.indexer.bitcoin
+
+import java.util.concurrent.Executors
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import cats.effect.{ContextShift, ExitCode, IO, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import org.http4s.headers.Authorization
+import org.http4s.BasicCredentials
+import org.http4s.client.blaze.BlazeClientBuilder
+import slick.jdbc.PostgresProfile.api._
+import slickeffect.Transactor
+import slickeffect.transactor.config
+
+import tech.cryptonomic.conseil.common.util.DatabaseUtil
+import tech.cryptonomic.conseil.indexer.config.LorreConfiguration
+import tech.cryptonomic.conseil.common.config.Platforms
+import tech.cryptonomic.conseil.common.config.Platforms.BitcoinConfiguration
+import tech.cryptonomic.conseil.indexer.LorreIndexer
+import tech.cryptonomic.conseil.indexer.logging.LorreProgressLogging
+import tech.cryptonomic.conseil.common.rpc.RpcClient
+
+/**
+  * Class responsible for indexing data for Bitcoin Blockchain
+  *
+  * @param lorreConf Lorre configuration
+  * @param bitcoinConf Bitcoin configuration
+  */
+class BitcoinIndexer(
+    lorreConf: LorreConfiguration,
+    bitcoinConf: BitcoinConfiguration
+) extends LazyLogging
+    with LorreIndexer
+    with LorreProgressLogging {
+
+  private val executor = Executors.newFixedThreadPool(16)
+  implicit private val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.fromExecutor(executor))
+  implicit val timer = IO.timer(ExecutionContext.global)
+  implicit private val httpEC: ExecutionContext = ExecutionContext.fromExecutor(executor) // Implicit is used to provide ExecutionContext for `stop`
```

Comment:
Why two Ex.Contexts from the same executor, and not defining one and passing it to the ContextShift?

Reply:
I've changed this code, please review again